scripts/edl_file_upload.py

```diff
@@ -1,202 +1,259 @@
 #!/usr/bin/env python3
 """
 File upload to OreSat.
-
-Order of PDUs:
-  - [send] metadata
-  - loop:
-      - [send] data transfer
-      - [recv] possible nak
-  - [send] EOF
-  - [recv] EOF ack
-  - [recv] Finished
-  - [send] Finished ack
 """
 
-import os
 import random
+import signal
 import socket
 import sys
-import zlib
+import time
 from argparse import ArgumentParser
-from enum import IntEnum, auto
-from threading import Thread
-from time import sleep
-
-from olaf import OreSatFile
-from spacepackets.cfdp import CrcFlag
-from spacepackets.cfdp.conf import LargeFileFlag, PduConfig
-from spacepackets.cfdp.defs import ChecksumType, ConditionCode, Direction, TransmissionMode
-from spacepackets.cfdp.pdu import (
-    AckPdu,
-    DirectiveType,
-    EofPdu,
-    FinishedPdu,
-    MetadataParams,
-    MetadataPdu,
-    NakPdu,
-    TransactionStatus,
+from datetime import timedelta
+from pathlib import Path
+from queue import Empty, SimpleQueue
+from threading import Lock, Thread
+from typing import Any
+
+from cfdppy import CfdpState, PacketDestination, get_packet_destination
+from cfdppy.handler.dest import DestHandler
+from cfdppy.handler.source import SourceHandler
+from cfdppy.mib import (
+    CheckTimerProvider,
+    DefaultFaultHandlerBase,
+    IndicationCfg,
+    LocalEntityCfg,
+    RemoteEntityCfg,
+    RemoteEntityCfgTable,
+)
+from cfdppy.request import PutRequest
+from cfdppy.user import (
+    CfdpUserBase,
+    FileSegmentRecvdParams,
+    MetadataRecvParams,
+    TransactionFinishedParams,
+    TransactionId,
+    TransactionParams,
 )
-from spacepackets.cfdp.pdu.file_data import FileDataParams, FileDataPdu
+from olaf import OreSatFile
+from spacepackets.cfdp import CfdpLv
+from spacepackets.cfdp.defs import ChecksumType, ConditionCode, TransmissionMode
+from spacepackets.cfdp.tlv import ProxyPutRequest, ProxyPutRequestParams
+from spacepackets.countdown import Countdown
+from spacepackets.seqcount import SeqCountProvider
 from spacepackets.util import ByteFieldU8
 
-sys.path.insert(0, os.path.abspath(".."))
```

Comment:
Please keep this line

Reply:
I understand why this is here, but the proper solution to what this is trying to fix is a venv and `pip install --editable`